pkg/experiment/ingester/segment.go

```diff
@@ -513,9 +520,36 @@ func (s *segment) ingest(tenantID string, p *profilev1.Profile, id uuid.UUID, la
 	// CountReceivedBytes is tracked in distributors.
 }
 
+// getFunctionById returns a function by its ID
+func getFunctionById(p *profilev1.Profile, id uint64) (*profilev1.Function, bool) {
+	for _, fn := range p.Function {
+		if fn.Id == id {
+			return fn, true
+		}
+	}
+	return nil, false
+}
+
+// isHexAddress checks if a string looks like a hexadecimal memory address
+func isHexAddress(s string) bool {
+	// Most hex addresses in profiles start with "0x"
+	if len(s) < 3 || !strings.HasPrefix(s, "0x") {
+		return false
+	}
+
```

Comment:
are these functions used or needed? getFunctionById isHexAddress

Reply:
not anymore, good catch 👍